### PR TITLE
Add Pinata upload method

### PR DESCRIPTION
**Pull request by febo**
This PR adds Pinata as one of the upload methods. The configuration includes:
```rust
pub struct PinataConfig {
    pub jwt: String,
    pub api_gateway: String,
    pub content_gateway: Option<String>,
    pub parallel_limit: Option<u16>,
}
```
If the `content_gateway` is not specified, the `api_gateway` is used to generate the asset's URIs. The optional `parallel_limit` allows users to limit the number of parallel uploads according to their gateway rate limits.

### CODE REVIEW

**Comment by polluterofminds** [comment]
Fully support this! Let me or anyone on the Pinata team know if there are any Pinata-specific questions that can be answered to help move this PR along. Thanks for opening this @febo.

File: src/create_config/process.rs [hunk 1]
```diff
@@ -656,6 +658,40 @@ pub fn process_create_config(args: CreateConfigArgs) -> Result<()> {
...
 
+    if config_data.upload_method == UploadMethod::Pinata {
+        let jwt: String = Input::with_theme(&theme)
+            .with_prompt("What is your Pinata JWT authentication?")
+            .interact()
+            .unwrap();
+
+        let api_gateway = Input::with_theme(&theme)
+            .with_prompt("What is the Pinata gateway for upload?")
+            .default(String::from("default"))
```
Comment (samuelvanderwaal) [inline src/create_config/process.rs:669]: Looks like this is missing an actual default value.

**Review by samuelvanderwaal — CHANGES_REQUESTED** [review]
This will need docs changes added [here](https://docs.metaplex.com/developer-tools/sugar/reference/upload-methods) and [here](https://docs.metaplex.com/developer-tools/sugar/reference/configuration). 

When I use `https://api.pinata.cloud` gateways I get "invalid routes" on my generated links. [Example link](https://api.pinata.cloud/ipfs/QmX2S2tQFqootw4Sv56FKvtGRTzpD8KqKQx1SsYfwgs8Lt/3.json). Not sure if my free API endpoint is insufficient or I'm doing the setup wrong.

**Comment by febo** [comment]
The public gateway is https://gateway.pinata.cloud. The upload worked for [the example](https://gateway.pinata.cloud/ipfs/QmX2S2tQFqootw4Sv56FKvtGRTzpD8KqKQx1SsYfwgs8Lt/3.json).

I will add the docs and fix the create config default value - thanks!

File: src/create_config/process.rs (hunk 1 above)
Comment (febo) [reply src/create_config/process.rs]: Fixed both default values - thanks!

**Comment by febo** [comment]
Changes made - PR for the docs created: https://github.com/metaplex-foundation/docs/pull/349

**Review by samuelvanderwaal — APPROVED** [review]

**Pull request merged**